game/state/playerstatesnapshot.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "game/state/gamestate.h"
+#include "game/state/gametime.h"
+#include "game/state/shared/organisation.h"
+#include "library/strings.h"
+#include "library/sp.h"
+
+namespace OpenApoc
+{
+
+class PlayerStateSnapshot : public std::enable_shared_from_this<PlayerStateSnapshot>
```

Comment:
As this class is relatively small and copy able, it might make sense to avoid using shared ptr for access and just copy as needed? I want to avoid using shared_ptr as much as possible, as it often ends up being a crutch instead of actually thinking about ownership of the data